packages/hardhat-plugin/test/verify.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable import/no-unused-modules */
+import type { Etherscan } from "@nomicfoundation/hardhat-verify/etherscan";
+
+import { assert } from "chai";
+
+import { getApiKeyAndUrls } from "../src/utils/getApiKeyAndUrls";
+import { verifyEtherscanContract } from "../src/utils/verifyEtherscanContract";
+
+import { useEphemeralIgnitionProject } from "./use-ignition-project";
+
+describe("verify", function () {
+  describe("when there is no etherscan API key configured", function () {
+    useEphemeralIgnitionProject("verify-no-api-key");
+
+    it("should throw", async function () {
+      await assert.isRejected(
+        this.hre.run(
+          { scope: "ignition", task: "verify" },
+          {
+            deploymentId: "test",
+          }
+        ),
+        /No etherscan API key configured/
+      );
+    });
+  });
+
+  describe("utils", function () {
+    describe("getApiKeyList", function () {
+      it("should return the correct API URLs when given a string", function () {
+        const apiKeyList = getApiKeyAndUrls("testApiKey", "mainnet");
+
+        assert.deepEqual(apiKeyList, [
+          "testApiKey",
+          "https://api.etherscan.io/api",
+          "https://etherscan.io",
+        ]);
+      });
+
+      it("should return the correct API URLs when given an object and a configured network name", function () {
+        const apiKeyList = getApiKeyAndUrls(
+          {
+            goerli: "goerliApiKey",
+            sepolia: "sepoliaApiKey",
+          },
+          "goerli"
+        );
+
+        assert.deepEqual(apiKeyList, [
+          "goerliApiKey",
+          "https://api-goerli.etherscan.io/api",
+          "https://goerli.etherscan.io",
+        ]);
+      });
+
+      it("should throw when given a string if the network name is not mainnet", function () {
+        assert.throws(() => getApiKeyAndUrls("testApiKey", "goerli"));
+      });
+
+      it("should throw when given an object and a nonexistent network name", function () {
+        assert.throws(
+          () =>
+            getApiKeyAndUrls(
+              {
+                goerli: "goerliApiKey",
+                sepolia: "sepoliaApiKey",
+              },
+              "mainnet"
+            ),
+          /No etherscan API key configured for network mainnet/
+        );
+      });
+    });
+
+    describe("verifyEtherscanContract", function () {
```

Comment:
What happens in the case where the bytecode has been verified in a previous verification (say the constructor args varied but the contract deploy bytecode is the same?).

Reply:
Etherscan will auto-verify any verified bytecode that it recognizes and will return the `Already Verified` message that the `verify` task handles internally